test-clients/src/main/java/com/hedera/services/bdd/suites/contract/SystemDeleteContractSuite.java

```diff
@@ -0,0 +1,98 @@
+package com.hedera.services.bdd.suites.contract;
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.HapiSpecSetup;
+import com.hedera.services.bdd.spec.utilops.UtilVerbs;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.hedera.services.bdd.spec.HapiApiSpec.defaultHapiSpec;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getContractInfo;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.systemContractDelete;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.systemContractUndelete;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+
+public class SystemDeleteContractSuite extends HapiApiSuite {
```

Comment:
I think you can add these tests to contractDeleteSuite instead of creating a new test class.

Reply:
This is a good question. I've been thinking how to organize the test suites for some time. To me the big challenge of the test suite organization is to have biggest coverage with shortest test running time. To achieve this, the key is to reduce the redundancy. This will naturally lead people to put all test cases in one suite. This of course is unacceptable. So how to dice and slice the test cases is a good topic we need to discuss.
 
My current tentative thought is to put all ordinary small contract operations (Create/Read(query)/Update/Delete)  that are used by normal users day in day out into one suite, instead of separated by CRUD operations.  This one suite goes into CI workflow.  In addition to this, we will have several suites to cover more advanced, special corner cases  Any system account involved test cases should go into one or more suites of the advanced coverage suites. And these test suites will go to nightly regression workflow. 

File, crypto, consensus, token and any future services can follow same organization.

One issue with this is that it's not easy for people to know what test cases we already have. If a developer plans to add a new coverage, he or she may need to go through quite some test suites to find out if the test cases already exist.

I guess this will be the first step to move towards this direction.